src/main/java/org/obiba/es/mica/query/RQLQuery.java

```diff
@@ -185,11 +185,11 @@ private void parseNode(ASTNode node) {
               case FIELDS:
                 parseFields(n);
                 break;
-              case MATCH:
-                if (n.getArgumentsSize() == 1) {
-                  parseFullTextMatch(n);
-                } else {
-                  parseQuery(n);
+              case FILTER: // for now will only have one argument RQL_NODE.MATCH
```

Comment:
Why is there such restriction?

Reply:
It's not a "fullTextMatchQuery" that you are building, but a generic filterQuery. Whether it is a full-text match query is a client concern.